srcpkgs/goldendict-ng/template

```diff
@@ -0,0 +1,22 @@
+# Template file for 'goldendict-ng'
+pkgname=goldendict-ng
+version=24.05.05
+revision=1
+_version_name=LiXia.ecd1138c
+build_style=cmake
+configure_args="-DUSE_ALTERNATIVE_NAME=ON"
+makedepends="qt6-base qt6-speech-devel qt6-svg-devel qt6-qt5compat-devel qt6-base-devel
+ qt6-webengine-devel qt6-multimedia-devel qt6-speech-plugin-flite
+ qt6-speech-plugin-mock qt6-speech-plugin-speechd pkg-config ffmpeg-devel
+ hunspell-devel opencc-devel libvorbis-devel xapian-core-devel libeb-devel
+ libzim-devel qt6-tools-devel"
```

Comment:
```suggestion
configure_args="-DUSE_ALTERNATIVE_NAME=ON -DUSE_SYSTEM_FMT=ON"
hostmakedepends="pkg-config qt6-declarative-host-tools qt6-tools"
makedepends="ffmpeg-devel fmt-devel hunspell-devel libeb-devel libvorbis-devel
 libzim-devel opencc-devel qt6-base-devel qt6-declarative-devel
 qt6-multimedia-devel qt6-qt5compat-devel qt6-svg-devel qt6-webengine-devel
 xapian-core-devel"
depends="qt6-svg"
```
- Use `fmt` library from the repos
- Move `pkg-config` to `hostmakedepends`
- Install required `qt6-*tools`, but only for the host, to avoid `Exec format error`
- Explicit `qt6-svg` dependency needed, or most icons are missing from the UI

Reply:
Maybe I'm overseeing something but I don't see Text-to-speech makedepends here. So I guess we either have to go with:
1) configure_args+=" -DWITH_TTS=OFF"
or
2) makedepends+=" qt6-speech-devel qt6-speech-plugin-flite qt6-speech-plugin-mock
 qt6-speech-plugin-speechd"

Either way, cross-compilation for aarch64 works!

Reply:
Also, post_install thing seems redundant as the pixmap is 644 already :
`-rw-r--r--  0 0      0       22251 Jul 23 21:42 ./usr/share/pixmaps/goldendict-ng.png
`

Reply:
You're right, pasted this from a build where I had disabled the speech stuff. Couldn't get it to work, even after installing the plugins (which would probably also need to go into `depends`). Upstream disabled TTS later by default (xiaoyifang/goldendict-ng#1414, [change](https://github.com/xiaoyifang/goldendict-ng/commit/bd194c0fbd268ae0498b824be0a570d7c30843fc)). We should probably go with `-DWITH_TTS=OFF`.

Reply:
@mvf by the way, are you sure `depends="qt6-svg"` is required? I just built the package with this commented out and no icons are missing from my UI.

Reply:
@slymattz Yeah, `qt6-svg` is a sneaky dep. I only caught this because I already knew from another package. If you want to try for yourself, you can put a file with `ignorepkg=qt6-svg` into `/etc/xbps.d`, uninstall `qt6-svg` and enjoy the broken UI. :sweat_smile:

Reply:
I took the liberty of taking the fixes/suggestions above and putting them into a template for review purposes. Good news is it builds for x86_64 and cross-builds for x86_64-musl, aarch64 and aarch64-musl just fine.

```
# Template file for 'goldendict-ng'
pkgname=goldendict-ng
version=24.05.05
revision=1
_version_name=LiXia.ecd1138c
build_style=cmake
configure_args="-DUSE_ALTERNATIVE_NAME=ON -DUSE_SYSTEM_FMT=ON -DWITH_TTS=OFF"
hostmakedepends="pkg-config qt6-declarative-host-tools qt6-tools"
makedepends="ffmpeg-devel fmt-devel hunspell-devel libeb-devel libvorbis-devel
 libzim-devel opencc-devel qt6-base-devel qt6-declarative-devel
 qt6-multimedia-devel qt6-qt5compat-devel qt6-svg-devel qt6-webengine-devel
 xapian-core-devel"
depends="qt6-svg"
short_desc="Feature-rich dictionary lookup program (goldendict fork)"
maintainer="RandallFlagg <ykpui@hotmail.com>"
license="GPL-3.0-or-later"
homepage="https://xiaoyifang.github.io/goldendict-ng/"
distfiles="https://github.com/xiaoyifang/goldendict-ng/archive/refs/tags/v${version}-${_version_name}.tar.gz"
checksum=e7cf55fef4ddf00d87b4e54489ec33a605b1d5634d0b7c46ddcdd261c3113790
```

There's only one problem I experience with the build but it might be upstream-related. So, I haven't been able to get ffmpeg backend to work, be it with ffmpeg installed on my machine or not. For me, it is only Qt Multimedia or an external program like vlc that makes the audio work.

@mvf - may I ask you to check whether ffmpeg works for you?

With the ffmpeg backend on and goldendict-ng opened in a console session, I get:

> QIODevice::seek (QIODevice): Cannot call seek on a sequential device
open link url: QUrl("gdau://dc14cdf10c4508af380f2ac5c43c0a0f/MOVIE.mp3")
getResource: "gdau://dc14cdf10c4508af380f2ac5c43c0a0f/MOVIE.mp3"
scheme: "gdau"
host: "dc14cdf10c4508af380f2ac5c43c0a0f"
Codec open: MP3 (MPEG audio layer 3): channels: 1, rate: 22050, format: fltp

Reply:
@slymattz FFmpeg works for me. Console output is the same, including the `Cannot call seek` error. In `pavucontrol`'s `Playback` tab, I can see either backend opening a `QtmPulseStream` channel. Qt Multimedia kills it immediately after playback, for FFmpeg it sticks around and is replaced in subsequent playbacks.

Reply:
Then it's probably my setup then. I see a lot of recent revbumps and rebuilds for ffmpeg6 anyway. Does that mean that this change should also be reflected here (ffmpeg-devel6 in place of ffmpeg-devel) (ping @classabbyamp)?

Reply:
yes, all new packages should use ffmpeg6

Reply:
OK. So the change is now reflected in https://github.com/void-linux/void-packages/pull/51071#discussion_r1694297671 @RandallFlagg let me ask you since you're the OP, are you still interested in pushing updates to this PR and maintaining the package in the future?

Reply:
@classabbyamp

Like I wrote above, OP hasn't contributed to this PR in 3 weeks already and I doubt they want to maintain this package. We've already figured out a template that cross-builds for aarch64 and aarch64-musl and my question is - do we have to wait it out until this PR becomes stale or is there a possibility to consider a commit from my fork as we speak? I don't distrohop so keeping the package up-to-date shouldn't be a problem for me. Here's [my commit](https://github.com/void-linux/void-packages/compare/master...slymattz:void-packages:goldendict-ng-24.05.05)

Reply:
I added the missing bit with broken host-target wordsizes to exclude the architectures which do not have qt6 webengine, as suggested by abby.